### PR TITLE
Remove extra output generated by emcc when running clang

### PR DESCRIPTION
**Pull request by sbc100**
- Output the compiler command line just one using the standard
  shared.print_compiler_stage mechanism.
- No need for special handling of failure, we have check_call for that
  purpose.
- Many tests can now use `-v` rather than EMCC_DEBUG to get the internal
  compiler command line.
- Remove old test_dashM_consistent and test_dashE_consistent. I looked
  at updating these tests but couldn't figure out what useful purpose
  they served.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1388,6 +1388,7 @@ def print_compiler_stage(cmd):
   before executing it."""
   if '-v' in COMPILER_OPTS:
     print(' "%s" %s' % (cmd[0], ' '.join(cmd[1:])), file=sys.stderr)
+    sys.stderr.flush()
```
Comment (kripken) [inline tools/shared.py]: why is this needed?

Comment (sbc100) [reply tools/shared.py]: Because otherwise the stderr will be buffered by default and the subprocess output will appear before this line reporting that we are about to run it.
